### PR TITLE
Add WFC3 notebook 'calwf3_recal_tvb.ipynb'

### PR DESCRIPTION
**Pull request by dulude**
**Relevant Tickets**
- [SPB-1619](https://jira.stsci.edu/browse/SPB-1619)

------
This notebook checklist has been made available to us by the [Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [ ] There is only one H1 (`#` in Markdown) used in the notebook.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] All acronyms are defined at least the first time they are used. 
- [ ] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [ ] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [ ] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [ ] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [ ] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

--------------------------------------------------------------------------------------------------------------------------

Fred's notes for wfc3_calwf3_recalibration_notebook:

- Removed notebook from _config
- Added notebook to _toc
- Small grammar update in README
- Added `crds` to requirements
- Added pre-requirements to install hstcal
- Updated notebook to PEP8 standards

### CODE REVIEW

**Comment by FDauphin** [comment]
@haticekaratay @camipacifici @dulude I was able to pass the PEP8 style check, but failed the Notebook Execution and Validation. 

- In the raw .ipynb file, the Python version under metadata is 3.8.12, but CI is "Set up" in Python 3.11. Should I update that to Python 3.11?
- Under "Execute notebooks", the CI returns a `FileExistsError` when I try to make a local directory named `orig/`. How can this subdirectory already exist?
- The final error is under "Checkout only the file to the target branch". If the other two issues above are resolved, will this also resolve?

I appreciate any help towards fixing these small bugs

**Comment by FDauphin** [comment]
> @haticekaratay @camipacifici @dulude I was able to pass the PEP8 style check, but failed the Notebook Execution and Validation.
> 
> * In the raw .ipynb file, the Python version under metadata is 3.8.12, but CI is "Set up" in Python 3.11. Should I update that to Python 3.11?
> * Under "Execute notebooks", the CI returns a `FileExistsError` when I try to make a local directory named `orig/`. How can this subdirectory already exist?
> * The final error is under "Checkout only the file to the target branch". If the other two issues above are resolved, will this also resolve?
> 
> I appreciate any help towards fixing these small bugs

Hi @haticekaratay @dulude , I saw in #100 that the `pre-requirements.sh` bug was resolved so I came back here to check if the CI ran again. There were some minor conflicts that I resolved and the CI ran again, with all checks passing. However, it still produces an error on my second bullet point. Why would it pass if it found an error? I appreciate all the help!

**Comment by haticekaratay** [comment]
> > @haticekaratay @camipacifici @dulude I was able to pass the PEP8 style check, but failed the Notebook Execution and Validation.
> > 
> > * In the raw .ipynb file, the Python version under metadata is 3.8.12, but CI is "Set up" in Python 3.11. Should I update that to Python 3.11?
> > * Under "Execute notebooks", the CI returns a `FileExistsError` when I try to make a local directory named `orig/`. How can this subdirectory already exist?
> > * The final error is under "Checkout only the file to the target branch". If the other two issues above are resolved, will this also resolve?
> > 
> > I appreciate any help towards fixing these small bugs
> 
> Hi @haticekaratay @dulude , I saw in #100 that the `pre-requirements.sh` bug was resolved so I came back here to check if the CI ran again. There were some minor conflicts that I resolved and the CI ran again, with all checks passing. However, it still produces an error on my second bullet point. Why would it pass if it found an error, and is it ok for review? I appreciate all the help!

Thanks, @FDauphin, for bringing to attention the issue of the CI showing all green checks despite errors. This issue has been reported to the CI team( @dulude, @mgough-970). Apart from that, I've provided some suggestions for enhancing the notebook. Thanks again for your work.

File: notebooks/WFC3/calwf3_recalibration/calwf3_recal_tvb.ipynb [hunk 1]
```diff
@@ -55,20 +55,18 @@
     "This notebook assumes you have created the virtual environment in [WFC3 Library's](https://github.com/spacetelescope/WFC3Library) installation instructions.\n",
```
Comment (haticekaratay) [inline notebooks/WFC3/calwf3_recalibration/calwf3_recal_tvb.ipynb:55]: <p>In reviewing the part of the notebook where files are being renamed and moved, I'd like to suggest a small enhancement for clarity. The code snippet uses <code>os.path.basename</code> and <code>os.path.join</code> to handle file paths which is great practice for cross-platform compatibility and code readability.  </p><p> </p><pre class="ql-syntax" spellcheck="false">for im in science_files:
&nbsp;&nbsp;root = os.path.basename(im)
&nbsp;&nbsp;new_path = os.path.join('.', root)
&nbsp;&nbsp;os.rename(im, new_path)

data_directory......
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (haticekaratay) [inline notebooks/WFC3/calwf3_recalibration/calwf3_recal_tvb.ipynb:55]: <p>Thanks for your work. I've reviewed the current approach for moving files, and I have a suggestion that might streamline the process, especially considering the potential for repeated runs of the notebook. To ensure that we efficiently handle files that may already exist in the destination directory, I propose the following code: </p><p> </p><pre class="ql-syntax" spellcheck="false">os.makedirs('orig/', exist_ok=True)

for file_pattern in ['ibohbf*_ima.fits', 'ibohbf*_flt.fits', 'ibohbf*_drz.fits']:
&nbsp;&nbsp;for file in glob.glob(file_pattern):
&nbsp;&nbsp;&nbsp;&nbsp;destination_path = os.path.join('orig', os.path.basename(file))
&nbsp;&nbsp;&nbsp;&nbsp;if os.path.isfile(destination_path):
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;os.remove(destination_path)
&nbsp;&nbsp;&nbsp;&nbsp;shutil.move(file, destination_path)&nbsp;&nbsp;
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

**Comment by FDauphin** [comment]
@haticekaratay Thanks for the comments. I've integrated them into the notebook. Now the notebook execution and validation test completes with no errors. However, now I have an error in the Greetings check, which I am unfamiliar with. What should I change so it passes? Thanks for the help.

**Review by haticekaratay — APPROVED** [review]
Thanks for your work @FDauphin. The Action `greetings` can be ignored for our case, and it's been reported to the CI team.

**Pull request merged**